File: src/app-server/csv-random.js
function csv_random(x)
{
 check(is_obj,x)
 
 function get_signal(x)
 { 
  check(is_uint,x)
  
  const r=arr()
  const base=random(8192)
  const range=random(4096)
  
  for(let i=0;i<x;i++)
  {
   const half=div(range,2)
    
   let v=random(range)
    
   v=add(v,base)
   v=sub(v,half)
   v=trunc(v)
    
   push(r,v)
  }
  
  return r
 }
 
 const r=arr()
 const length=random(58)
 const signals=arr()
 
 for(let i=0;i<dimension;i++)
 {
  const signal=get_signal(length)
  
  push(signals,signal)
 }
 
 for(let i=0;i<length;i++)
 {
  const row=arr()
  
  for(let j=0;j<dimension;j++)
  {
   const signal=at(signals,j)
   const n=at(signal,i)
   
   push(row,n)
  }
  
  push(r,row) 
 }
 
 return r
}

File: src/prj-jneeg/app-server-jneeg/csv-random.cs
fn csv_random x:obj
 //get signal

 fn get_signal x:uint
  let r arr
  let base random 8192
  let range random 4096

  fornum x
   let half div range 2
   let v random range
   let v add v base
   let v sub v half
   let v trunc v

   push r v
  end

  ret r
 end

 //main

 let r arr
 let length random 58
 let signals arr

 fornum dimension
  let signal get_signal length

  push signals signal
 end

 fornum length
  let row arr
  let index i

  fornum dimension
   let signal at signals i
   let n at signal index

   push row n
  end

  push r row
 end

 ret r
end